### PR TITLE
Feat/new search hit card - LESQ-459

### PR DESCRIPTION
**Pull request by JBR90**
## Description

- Adds new search hit card
- Decouples search hits from lesson list and unit list (helpers/types)
- Decouple search tracking from components and add at page level
- Updates/adds tests
- Adds to storybook
- Adds new tag to search hits
- Fixes pagination padding 

Music year: {owa_music_year}

## Acceptance Criteria:

- [x]  New component for search
- [x]  New component shows all details on result (type, key stage, exam boards etc) (no exam or year data yet - LESQ-486)
- [x]  Lesson is pink circle, unit is blue square
- [x]  Units just have title, lessons have title and description
- [x]  See lesson/see unit button
- [x]  Whole thing clickable (for now)
- [x]  Lesson metadata component should be able to handle multiple exam boards
- [x]  search results page is using this new card

## Issue(s)

Fixes #

## How to test

1. Go to https://deploy-preview-2114--oak-web-application.netlify.thenational.academy/teachers/search?term=waves&flag=enable_2023_search_api

Check new and legacy hits take you to the correct page
Tracking searchOnClick event should be recorded 

## Screenshots

How it used to look (delete if n/a):
{screenshots}

How it should now look:
{screenshots}

## Checklist

- [ ] Added or updated tests where appropriate
- [ ] Manually tested across browsers / devices
- [ ] Considered impact on accessibility
- [ ] Design sign-off
- [ ] Approved by product owner
- [ ] Does this PR update a package with a breaking change

### CODE REVIEW

**Review by jack-skerrett-oak — CHANGES_REQUESTED** [review]
- I confused myself checking on the designs because everything is labelled "Lesson", but I think currently the lesson/unit cards are the wrong way round. In our browse journey, units are blue and lessons are pink. 
- The "see lesson" links are lacking the focus styling they have in figma`

I will continue reviewing after this meeting!

**Review by Monika-Koziol — CHANGES_REQUESTED** [review]
Coming along really well. A few small changes if possible:

- [x] Use grey-60 for the meta content (lesson, key stage, etc,)
- [ ] There is a bigger gap between card component and filters. In figma I left one column width. Or the card can have a max-width 734px?

**Review by Monika-Koziol — COMMENTED** [review]
Looking great. Not sure if this is part of this ticket but now the new search bar got merged would it be possible for the `search bar+button` to be the same width as the search results card?

File: src/components/SearchResultsItem/SearchResultsItem.test.tsx [hunk 1]
```diff
@@ -0,0 +1,180 @@
...
+        <SearchResultsItem
+          {...hitObjectLesson}
+          onClick={searchResultClicked}
+        />,
+      );
+      const link = getByText("Dipping into Macbeth - Brave Macbeth (Part 2)");
+      act(() => {
+        link.click();
+      });
+      expect(searchResultClicked).toHaveBeenCalled();
```
Comment (k-huggs) [inline src/components/SearchResultsItem/SearchResultsItem.test.tsx:175]: Is it worth testing what this should be called with at the same time?

**Review by k-huggs — APPROVED** [review]
One small comment on testing, but LGTM ✅

**Review by jack-skerrett-oak — APPROVED** [review]
Looks great. I like what you have done with the area labels on the search results :)

**Review by jack-skerrett-oak — CHANGES_REQUESTED** [review]
Actually I have just noticed something on iPhone on browser stack. The bullet between the type and key stage gets very squished 
Screenshot 2023-11-29 at 08 51 19

**Review by jack-skerrett-oak — APPROVED** [review]
OK, that might have just been a browser stack issue

**Pull request merged**